packages/android_alarm_manager_plus/android/src/main/java/dev/fluttercommunity/plus/androidalarmmanager/AlarmService.java

```diff
@@ -147,7 +148,19 @@ private static void scheduleAlarm(
       if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S && !manager.canScheduleExactAlarms()) {
         Log.e(TAG, "Can`t schedule exact alarm due to revoked SCHEDULE_EXACT_ALARM permission");
       } else {
-        AlarmManagerCompat.setAlarmClock(manager, startMillis, pendingIntent, pendingIntent);
+        PackageManager packageManager = context.getPackageManager();
+        String appId = context.getPackageName();
+        Intent launchIntent = packageManager.getLaunchIntentForPackage(appId);
+        launchIntent.putExtra("id", requestCode);
+        launchIntent.putExtra("params", params == null ? null : params.toString());
+        PendingIntent showPendingIntent =
+            PendingIntent.getActivity(
+                context,
+                requestCode,
+                launchIntent,
+                (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M ? PendingIntent.FLAG_IMMUTABLE : 0)
+                    | PendingIntent.FLAG_UPDATE_CURRENT);
```

Comment:
Not a big deal, but could you move the `showPendingIntent` construction code into a separate function?